crates/matrix-sdk-base/src/read_receipts.rs

```diff
@@ -166,103 +412,99 @@ impl PreviousEventsProvider for () {
 /// See this module's documentation for more information.
 ///
 /// Returns a boolean indicating if a field changed value in the read receipts.
-#[instrument(skip_all, fields(room_id = %room_id, ?read_receipts))]
-pub(crate) fn compute_notifications<PEP: PreviousEventsProvider>(
+#[instrument(skip_all, fields(room_id = %room_id))]
+pub(crate) fn compute_notifications(
     user_id: &UserId,
     room_id: &RoomId,
     receipt_event: Option<&ReceiptEventContent>,
-    previous_events_provider: &PEP,
+    previous_events: Vector<SyncTimelineEvent>,
     new_events: &[SyncTimelineEvent],
     read_receipts: &mut RoomReadReceipts,
 ) -> Result<bool> {
-    let prev_latest_receipt_event_id = read_receipts.latest_read_receipt_event_id.clone();
+    debug!(?read_receipts, "Starting.");
+
+    let previous_events_ids =
+        BTreeSet::from_iter(previous_events.iter().filter_map(|ev| ev.event_id()));
+
+    let all_events = if new_events
+        .iter()
+        .any(|ev| ev.event_id().map_or(false, |event_id| previous_events_ids.contains(&event_id)))
```

Comment:
Could this be a named function? I found it difficult to spot what it did immediately. Maybe `events_intersect`?